File: 48/rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int N = matrix.length;
        int[][] temp = new int[N][N];
        for(int i=0; i<N; i++) {
            for(int j=0; j<N; j++) {
                temp[i][j] = matrix[N-1-j][i];
            }
        }
        for(int i=0; i<N; i++){
            for(int j=0; j<N; j++){
                matrix[i][j] = temp[i][j];
            }
        }
    }
}

File: 48/rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int N = matrix.size();
        for(int i=0; i<N/2; i++) {
            for(int j=0; j<(N+1)/2; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[N-1-j][i];
                matrix[N-1-j][i] = matrix[N-1-i][N-1-j];
                matrix[N-1-i][N-1-j] = matrix[j][N-1-i];
                matrix[j][N-1-i] = temp;
            }
        }
    }
};
